cinema/views.py

```diff
@@ -31,9 +34,32 @@ class CinemaHallViewSet(viewsets.ModelViewSet):
 
 
 class MovieViewSet(viewsets.ModelViewSet):
-    queryset = Movie.objects.all()
+    queryset = Movie.objects.all().prefetch_related("actors", "genres")
     serializer_class = MovieSerializer
 
+    def get_queryset(self):
+        actors = self.request.query_params.get("actors")
+        genres = self.request.query_params.get("genres")
+        title = self.request.query_params.get("title")
+        queryset = self.queryset
+
+        if actors:
+            actors_ids = [int(str_id) for str_id in actors.split(",")]
+            queryset = queryset.filter(
+                actors__id__in=actors_ids
+            ).prefetch_related("actors", "genres")
```

Comment:
you could probably get on with 1 prefetch related if you would do this in    " queryset = self.queryset" part instead of both for actors and genres

Reply:
good point